addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -150,8 +150,9 @@ def create(self, vals):
 
     def write(self, vals):
         res = super().write(vals)
-        if not self.picking_ids and self.state == 'in_progress':
-            self.action_cancel()
+        for batch in self:
+            if not batch.picking_ids and batch.state == 'in_progress':
+                batch.action_cancel()
```

Comment:
small performance remark, you can call action_cancel once on all batches that need to be cancelled

Reply:
Ah I thought the same thing, but action_cancel has a `self.ensure_one()` in it. Any idea why it's there? And if it's safe to remove in stable?

Reply:
it has self.ensure_one() 😢

Reply:
I'm surprised we still have action_* with ensure_one(). We should make it 'multi'